src/v/datalake/local_parquet_file_writer.cc

```diff
@@ -195,7 +195,11 @@ local_parquet_file_writer_factory::create_writer(
     // of too many writers, needs empirical evaluation to determine the correct
     // sizing.
     static constexpr size_t WRITER_RESERVATION_OVERHEAD = 10_KiB;
-    co_await _mem_tracker.reserve_bytes(WRITER_RESERVATION_OVERHEAD, as);
+    auto reservation_err = co_await _mem_tracker.reserve_bytes(
```

Comment:
nit: if this is no longer a draft it'd be nice if each non-trivial commit had a meaningful commit message